### ISSUE TITLE
Why is it so slow?

### ISSUE THREAD

**Issue by JesseLT** [tensorflow/tensorflow#50718]
I'm using MacBook Air with M1 chip. OS version is Big Sur 11.4.
`which python`
`/Users/dmitry/Applications/Miniforge3/bin/python`
I run the following code using [tensorflow-macos](https://developer.apple.com/metal/tensorflow-plugin/) and [tensorflow_macos](https://github.com/apple/tensorflow_macos), respectively.
`import tensorflow as tf`
`mnist = tf.keras.datasets.mnist`
`(x_train, y_train), (x_test, y_test) = mnist.load_data()`
`x_train, x_test = x_train / 255.0, x_test / 255.0`
`model = tf.keras.models.Sequential([`
 `     tf.keras.layers.Flatten(input_shape=(28, 28)),`
 `     tf.keras.layers.Dense(128, activation='relu'),`
 `     tf.keras.layers.Dropout(0.2),`
 `     tf.keras.layers.Dense(10)`
`])`
`predictions = model(x_train[:1]).numpy()`
`tf.nn.softmax(predictions).numpy()`
`loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)`
`loss_fn(y_train[:1], predictions).numpy()`
`model.compile(optimizer = 'sgd', loss = loss_fn)`
`model.fit(x_train, y_train, epochs=100)`

I got this! with [tensorflow-macos](https://developer.apple.com/metal/tensorflow-plugin/) and python3.9:
1875/1875 [==============================] - 8s 4ms/step - loss: 0.7026
Epoch 2/100
1875/1875 [==============================] - 8s 4ms/step - loss: 0.3872
Epoch 3/100
1875/1875 [==============================] - 8s 4ms/step - loss: 0.3284
Epoch 4/100
1875/1875 [==============================] - 8s 4ms/step - loss: 0.2891
Epoch 5/100
1875/1875 [==============================] - 8s 4ms/step - loss: 0.2622

with [tensorflow_macos](https://github.com/apple/tensorflow_macos) and python3.8 env:
Epoch 1/100
1875/1875 [==============================] - 1s 276us/step - loss: 1.2181
Epoch 2/100
1875/1875 [==============================] - 1s 270us/step - loss: 0.4678
Epoch 3/100
1875/1875 [==============================] - 1s 269us/step - loss: 0.3935
Epoch 4/100
1875/1875 [==============================] - 1s 271us/step - loss: 0.3507
Epoch 5/100
1875/1875 [==============================] - 1s 270us/step - loss: 0.3231

Why [tensorflow-macos](https://developer.apple.com/metal/tensorflow-plugin/) is so slower than [tensorflow_macos](https://github.com/apple/tensorflow_macos)?Did I miss something?

**Comment by tilakrayal** [comment]
@JesseLT ,

Can you please reduce the number of epochs and try to execute the code in new environment and let us know if you are facing same issue?Thanks!

**Comment by JesseLT** [comment]
@tilakrayal, I change the number of epochs to 10, and it has the same issue.

**Comment by tilakrayal** [comment]
@Saduf2019 ,
I was able to reproduce the issue in tf v2.4 and v2.5.Please find the gist [here](https://colab.research.google.com/gist/tilakrayal/a083a178028c1b34c7b991001b62a9a2/untitled50718.ipynb).

**Comment by Saduf2019** [comment]
@JesseLT 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more refer to:
https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999

**Comment by Saduf2019** [comment]
Moving this to closed status as issue is opened in keras.
